tests/test_run_reference_vmm.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Run the reference VMM."""
+
+import os, signal, subprocess, time
+import pytest
+
+
+def test_reference_vmm():
+    """Start the reference VMM and trust that it works."""
+
+    # Memory config
+    mem_size_mib = 1024
+
+    # Kernel config
+    cmdline = "console=ttyS0 i8042.nokbd reboot=k panic=1 pci=off"
+    kernel_path = os.path.abspath(os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        "..",
+        "resources/kernel/vmlinux-hello-busybox"
+    ))
+    himem_start = 1048576
+
+    # vCPU config
+    num_vcpus = 1
+
+    vmm_cmd = [
+        "cargo", "run", "--",
+        "--memory", "mem_size_mib={}".format(mem_size_mib),
+        "--kernel", "cmdline=\"{}\",path={},himem_start={}".format(
+            cmdline, kernel_path, himem_start
+        ),
+        "--vcpus", "num_vcpus={}".format(num_vcpus)
+    ]
+    
+    # We can't talk to the reference VMM yet, and it can't talk back.
+    # If we try to capture the output, Python doesn't return control to the
+    # test until the child process ends, which it doesn't. So we have to trust
+    # that the output is there, let the VMM run for a bit, then kill it.
+    # In the future, it will communicate via metrics / devices.
+    vmm_process = subprocess.Popen(vmm_cmd)
```

Comment:
Hmmm, are we sure that this test actually runs `vmm_cmd` successfully? `subprocess.Popen` seems to have a weird behavior (i.e. it fails, from what I noticed, only if the binary doesn't exist (in this case `cargo`, which exists)).
Wouldn't it be better to run `subprocess.run` for example or something else?

Also, funny question: is the `cargo run ...` option still valid with the workspace changes that I mentioned in another comment? I didn't manage to run it successfully now, but I might be doing something really wrong :-?.

Reply:
Unlike `Popen`, `subprocess.run` doesn't end until the child process ends - so the test would be stuck in it.
As for the workspace stuff, I'm looking into those, it's not clear to me yet how we want to proceed.

Reply:
From what I manually tested, the things mentioned here: https://github.com/rust-vmm/rust-vmm-ci/issues/38 seem to fix the workspace issue.

Reply:
I checked again, the test is still passing if, for example, we replace `run` with `blah` in `vmm_cmd`. In this case, the test is not useful. I'll try looking more into how we can make this test fail if the command is invalid.

Reply:
I think testing `cargo blah` instead of `cargo run` tests `cargo` more than it does the VMM :thinking:

Reply:
It was just an example, replacing `--memory` with some invalid arg will keep the test passing :(. In my opinion we should make sure that whatever `subprocess` function we choose, the test is passing only if `cargo run ...` (or even `cargo blah`) is successful.